src/panels/lovelace/common/handle-action.ts

```diff
@@ -70,29 +77,72 @@ export const handleAction = async (
         fireEvent(node, "hass-more-info", {
           entityId: config.entity ? config.entity : config.camera_image!,
         });
+      } else {
+        showToast(node, {
+          message: hass.localize(
+            "ui.panel.lovelace.cards.actions.no_entity_more_info"
+          ),
+        });
+        forwardHaptic("failure");
       }
       break;
     }
     case "navigate":
       if (actionConfig.navigation_path) {
         navigate(node, actionConfig.navigation_path);
+      } else {
+        showToast(node, {
+          message: hass.localize(
+            "ui.panel.lovelace.cards.actions.no_navigation_path"
+          ),
+        });
+        forwardHaptic("failure");
       }
       break;
     case "url": {
       if (actionConfig.url_path) {
         window.open(actionConfig.url_path);
+      } else {
+        showToast(node, {
+          message: hass.localize("ui.panel.lovelace.cards.actions.no_url"),
+        });
+        forwardHaptic("failure");
       }
       break;
     }
     case "toggle": {
       if (config.entity) {
-        toggleEntity(hass, config.entity!);
-        forwardHaptic("light");
+        if (DOMAINS_TOGGLE.has(computeDomain(config.entity))) {
```

Comment:
I dont think we should do this, we now already show a toast if the service is not found I think, and this just limits the user

Reply:
Yes, but that is quite misleading since the user specified `toggle` and not `call-service`. IMHO user should not have to know that under the hood both trigger a service call. That is why I went with a dedicated toast.

Reply:
Bram raised that point that some domains can turn on but not off (e.g. scenes). That would then raise a breaking error unless the users adjusted to use `call-service` => check removed.